pkg/mongoproxy/plugins/schema/types.go

```diff
@@ -234,35 +234,59 @@ func (c *Collection) ValidateUpdate(ctx context.Context, obj bson.D, upsert bool
 	if !c.EnforceSchema {
 		return nil
 	}
-
-	for _, e := range obj {
-		switch e.Key {
-		case "$currentDate", "$inc", "$min", "$max", "$mul":
-			if setFields == nil {
-				setFields = e.Value.(bson.D).Map()
-			} else {
-				for _, item := range e.Value.(bson.D) {
-					setFields[item.Key] = item.Value
+	logrus.Debugf("pending validation object: %s", obj)
+	/*
+		update command support ,
+		* A document that contains update operator expressions,
+		* A replacement document with only <field1>: <value1> pairs
+		https://www.mongodb.com/docs/v4.2/reference/command/update/#update-statement-documents
+	*/
+	op_map := BuildUpdateOpSet()
+	if !strings.HasPrefix(obj[0].Key, "$") || !SetContain(op_map, obj[0].Key) {
```

Comment:
isn't every key in op_map start with '$'?  what's the point of the prefix check here?